src/app/CommandSender.h

```diff
@@ -287,12 +286,18 @@ class CommandSender final : public Messaging::ExchangeDelegate
      */
     CommandSender(Callback * apCallback, Messaging::ExchangeManager * apExchangeMgr, bool aIsTimedRequest = false,
                   bool aSuppressResponse = false);
-    CommandSender(ExtendableCallback * apCallback, Messaging::ExchangeManager * apExchangeMgr, bool aIsTimedRequest = false,
-                  bool aSuppressResponse = false);
     CommandSender(std::nullptr_t, Messaging::ExchangeManager * apExchangeMgr, bool aIsTimedRequest = false,
                   bool aSuppressResponse = false) :
         CommandSender(static_cast<Callback *>(nullptr), apExchangeMgr, aIsTimedRequest, aSuppressResponse)
     {}
+    CommandSender(ExtendableCallback * apCallback, Messaging::ExchangeManager * apExchangeMgr, bool aIsTimedRequest = false,
+                  bool aSuppressResponse = false);
+    CommandSender(TestOnlyMarker aTestMarker, ExtendableCallback * apCallback, Messaging::ExchangeManager * apExchangeMgr,
```

Comment:
Should this be conditioned on CONFIG_BUILD_FOR_HOST_UNIT_TEST or so?

Reply:
https://github.com/project-chip/connectedhomeip/actions/runs/7908409985/job/21587522500?pr=32105

Some targets seemingly run `src/app/tests/TestCommandInteraction.cpp` without defining `CONFIG_BUILD_FOR_HOST_UNIT_TEST`

Reply:
No, that's silly.  I really wish we had an actually usable "only enable when doing unit tests" define, but I guess that should be a followup here.

Reply:
https://github.com/project-chip/connectedhomeip/issues/32138